From d0af8345ce86479bb5d8af04edd1ef201daad538 Mon Sep 17 00:00:00 2001
From: Filip Filmar <filmil@gmail.com>
Date: Fri, 15 Dec 2023 04:19:39 +0000
Subject: [PATCH 2/2] feat: copy the include files into the build dir

---
 Makefile | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Makefile b/Makefile
index 0f1a6ad..538d2d0 100644
--- a/Makefile
+++ b/Makefile
@@ -50,6 +50,7 @@ SUBMODULES 			= 	lib/libsdcc-z80
 all:	$(BUILD_DIR) $(SUBMODULES) $(SUBDIRS)
 	cp $(BUILD_DIR)/*.lib $(BIN_DIR)
 	cp $(BUILD_DIR)/$(CRT0).rel $(BIN_DIR)/$(CRT0)$(CRT0EXT)
+	cp --dereference -R $(ROOT)/include $(BIN_DIR)
 
 # "make install" is somewhat expected to be present.
 .PHONY: install
-- 
2.34.1

